client/src/pages/admin/MergeLocations.js

```diff
@@ -0,0 +1,516 @@
+                setFieldValue("approvalSteps", location.approvalSteps)
+              )}
+            />
+          </div>
+        </>
+      )}
+    </div>
+  )
+
+  function getActionButton(onClickAction) {
+    return (
+      <small>
+        <Button
+          icon={reversed ? "double-chevron-left" : ""}
+          rightIcon={reversed ? "" : "double-chevron-right"}
+          intent="primary"
+          text="Use value"
+          onClick={onClickAction}
+        />
+      </small>
+    )
+  }
+}
+
+LocationColumn.propTypes = {
+  location: PropTypes.instanceOf(Location),
+  setLocation: PropTypes.func.isRequired,
+  setFieldValue: PropTypes.func.isRequired,
+  reversed: PropTypes.bool,
+  style: PropTypes.object
+}
+
+LocationColumn.defaultProps = {
+  reversed: false
+}
+
+const LocationField = ({ label, value, align, reversed, action }) => {
+  return (
+    <div
+      style={{
+        borderBottom: "1px solid #CCCCCC",
+        display: "flex",
+        flexDirection: reversed ? "row-reverse" : "row",
+        justifyContent: "space-between",
+        alignItems: "center",
+        padding: "8px 0"
+      }}
+    >
+      <div style={{ flex: "1 1 auto" }}>
+        <div
+          style={{
+            fontWeight: "bold",
+            textAlign: align || "center",
+            textDecoration: "underline"
+          }}
+        >
+          {label}
+        </div>
+        <div style={{ textAlign: align || "center" }}>{value}</div>
+      </div>
+      {action}
+    </div>
+  )
+}
+
+LocationField.propTypes = {
+  label: PropTypes.string.isRequired,
+  value: PropTypes.node,
+  align: PropTypes.string,
+  reversed: PropTypes.bool,
+  action: PropTypes.node
+}
+
+const MergeLocations = ({ pageDispatchers }) => {
+  const history = useHistory()
+  const [saveError, setSaveError] = useState(null)
+  const [mergedLocation, setMergedLocation] = useState(new Location())
+  const [location01, setLocation01] = useState()
+  const [location02, setLocation02] = useState()
+
+  const setLocationFn01 = value => {
+    if (location02?.uuid && value?.uuid && location02.uuid === value.uuid) {
+      toast.error("Please select a different location!")
+    } else {
+      setMergedLocation(new Location())
+      setLocation01(null) // workaround for map already initialized error
+      if (value) {
+        setTimeout(() => setLocation01(new Location(value)))
+      }
+    }
+  }
+
+  const setLocationFn02 = value => {
+    if (location01?.uuid && value?.uuid && location01.uuid === value.uuid) {
+      toast.error("Please select a different location!")
+    } else {
+      setMergedLocation(new Location())
+      setLocation02(null) // workaround for map already initialized error
+      if (value) {
+        setTimeout(() => setLocation02(new Location(value)))
+      }
+    }
+  }
+
+  useBoilerplate({
+    pageProps: DEFAULT_PAGE_PROPS,
+    searchProps: DEFAULT_SEARCH_PROPS,
+    pageDispatchers
+  })
+
+  const map = useMemo(() => {
```

Comment:
I don't think there is a need to memoize a component

Reply:
I used it for performance optimization by avoiding unnecessary renders on Leaflet, but it works just fine without it. So I will remove it.